linq4j/src/main/java/org/apache/calcite/linq4j/tree/Primitive.java

```diff
@@ -384,6 +384,10 @@ static void checkRoundedRange(Number value, double min, double max) {
     }
   }
 
+  public static @Nullable Object integerCast(Primitive primitive, final Object value) {
```

Comment:
Q: Is this used via reflection? My IDE suggests that the method is unused.

Reply:
Not quite reflection. The way Calcite evaluates constant expressions is to generate a Java class with a Java method which is then compiled, loaded and executed at runtime.